### DISCUSSION TITLE
colouring multipe lines with multiple colors

### DISCUSSION THREAD

**Discussion by xalario** [jazznbass/scan#39] [Q&A] [answered]
Dear prof J. Wilbert

We are very happy to have found your website about "Analyzing single-case data with R and scan" (Analyzing single-case data with R and scan). Thank you for putting this resource online!

We are using the code for the preparation of a manuscript about language rehabilitation.

Despite the details you provide, we have stumbled upon an issue we could not solve.

We would like to draw a graph that includes both median and trend lines (or other lines, the point is: more than one). What we are not able to do is to draw the two lines in different colors, which is desired to increase readability. We are only able to choose a /single/ color for all lines.  (We are able to draw different colored lines on different graphs with the same data, but that takes extra space.)

This is what we have tried

1/ notice the vector c(“red”, “blue”) passed within the lines list:

    plot (P4_UNTR, style=newstyle,ylab = "correct_responses", case.names="P4", phase.names=c("Baseline", "Intervention", "Follow-up"), main="trend differences", lines=list('mean','trendA','trend',col=c(“red”, “blue”), lty= c('dotted','dashed'),lwd=4))

2/ trying col.lines with no better luck:

    plot (P4_UNTR, style=newstyle,ylab = "correct_responses", case.names="P4", phase.names=c("Baseline", "Intervention", "Follow-up"), main="trend differences", lines=list('mean','trendA','trend',col.lines=c(“red”, “blue”), lty= c('dotted','dashed'),lwd=4))

We were hoping you find time to provide some advice.

Thank you in advance. Best regards,

Véronique Sabadell & Xavier Alario

**Comment by jazznbass** [comment, marked as answer]
actually this feature is missing.

I will include it into the next scan update which is due within the next three weeks.


Would such a solution look "intuitive" to you?


lines = c(

  list(type = "mean", col = "red", lwd = 4, lty = "solid"),

  list(type = "trend", col = "blue", lwd = 2, lty = "dashed")

)

**Reply by xalario** [reply]
Thanks!! 
Yes, this looks intuitive from this side...: In fact we had tried our luck with something *very* similar

**Reply by jazznbass** [reply]
I have implemented the function into to development version 0.53

Example:

plot(exampleAB, lines = list(
  list("median", col = "red", lwd = 0.5),
  list("trend", col = "blue", lty = "dashed", lwd = 2),
  list("loreg", f = 0.2, col = "green", lty = "solid", lwd = 1)
  )
)

![image](https://user-images.githubusercontent.com/50012671/122384462-01e2f200-cf6c-11eb-8735-e98aabdba63d.png)

you can install this branch with:

devtools::install_github("jazznbass/scan", ref = "v0.53")

(make sure you have the package devtools installed in r and on a windows machine you need to install Rtools (rtools is not an r package)

Cheers!

**Comment by xalario** [comment]
We did not expect this reply so fast. It does work! thank you very much!!

One microscopic comment. We were testing whether a vector coding for multiple trend lines would work (it didn't of course). This was because we were thinking in the spirit of the `legend()` command:

`legend(1, 95, legend=c("Line 1", "Line 2"),
       col=c("red", "blue"), lty=1:2, cex=0.8)`

I suppose you use a `list()` object, rather than a vector, because of how the rest of the package is organized.

At any rate, thank you again!
